### PR TITLE
Sharareh #48/49

### PR DESCRIPTION
**Pull request by ShEnshaei**
This PR adds logic to update user password and to get user's info. Both endpoints leverage authorization and authentication.

### CODE REVIEW

File: api_auth/User/serializers.py [hunk 1]
```diff
@@ -49,3 +49,18 @@ def validate_phone_number(self, value):
             raise serializers.ValidationError('Phone number cannot be empty')
 
         return value
+
+class UserPasswordSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('password')
+
+    def validate_password(self, value):
```
Comment (pedroskzt) [inline api_auth/User/serializers.py]: Django has a builtin validate_password that we can configure with the desired rules. So, I guess we don't need to manually implement it. Check this for further info on that: [Django Documentation](https://docs.djangoproject.com/en/5.0/topics/auth/passwords/#module-django.contrib.auth.password_validation)

File: api_auth/urls.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
-from api_auth.User.UserView import (UserLogin, UserRegistration, UserUpdate, UserGetInfo)
+from api_auth.User.UserView import (UserLogin, UserRegistration, UserUpdate, UserGetInfo, ChangePassword)
 
 urlpatterns = [
     path('Login', UserLogin.as_view()),
     path('registration', UserRegistration.as_view()),
     path('User/Update', UserUpdate.as_view()),
     path('User/Get/Info/<int:user_id>', UserGetInfo.as_view())
+    path('ChangePassword/<int:user_id>', ChangePassword.as_view())
```
Comment (pedroskzt) [inline api_auth/urls.py]: If we are not going to use the user_id parameter, there's no need to keep it on the URL and in the code. (The same for User Get Info)

File: core/UserManager/UserHandler.py [hunk 3]
```diff
@@ -73,4 +73,47 @@ def handler_user_update(request_data, user):
...
+            return Response("Password must be passed in and cannot be null or empty", status = status.HTTP_400_BAD_REQUEST)
+
+        UserHelper.validate_password(request_data.get('password'))
+
+        if user:
+            user.set_password(request_data.get('password'))
+
+            return Response("Password has been updated successfully.", status=status.HTTP_200_OK)
+        else:
+            return Response(validated_data.errors, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```
Comment (pedroskzt) [inline core/UserManager/UserHandler.py]: Getting and errror on `validated_data.errors`, maybe you meant to use another variable.

File: core/UserManager/UserHelper.py [hunk 4]
```diff
@@ -6,3 +6,16 @@ class UserHelper:
...
         return User.objects.get(email__exact=email)
+
+    @staticmethod
+    def get_user_by_id(id):
+        return User.objects.get(id=id)
+
+    @staticmethod
+    def validate_password(password):
+        if len(password) <= 10:
+            raise serializers.ValidationError('Password must be greater than 10 characters')
```
Comment (pedroskzt) [inline core/UserManager/UserHelper.py]: Getting erros on both `serializers.ValidationError`. The library and the ValidateError class were not importedo.

Also, as I mentioned on previous comment, Django has a built-in password validation, it should be worth taking a look at it.

**Review by pedroskzt — CHANGES_REQUESTED** [review]

File: api_auth/urls.py (hunk 2 above)
Comment (ShEnshaei) [reply api_auth/urls.py:11]: nice catch, forgot to remove

**Review by pedroskzt — APPROVED** [review]

**Pull request merged**